### PR TITLE
Issue #SB-18620 allowing all kind of failed records to process once a…

### PR DESCRIPTION
**Pull request by Hari-stackroute**
…gain from the bulkupload

### CODE REVIEW

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ShadowUserMigrationScheduler.java [hunk 1]
```diff
@@ -231,19 +231,12 @@ private void updateUser(String processId,MigrationUser migrationUser, ShadowUser
      * ALSO this method will allow only CLAIMED user record to be updated in diksha db.
      * @param processId
      * @param migrationUser
-     * @param shadowUser
+     * @param existingShadowUser
```
Comment (indrajra) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ShadowUserMigrationScheduler.java:234]: The comments above require change.

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ShadowUserMigrationScheduler.java [hunk 2]
```diff
@@ -231,19 +231,12 @@ private void updateUser(String processId,MigrationUser migrationUser, ShadowUser
...
-            propertiesMap.put(JsonKey.EMAIL,migrationUser.getEmail());
-            propertiesMap.put(JsonKey.PHONE,migrationUser.getPhone());
-            propertiesMap.put(JsonKey.PROCESS_ID,processId);
-            propertiesMap.put(JsonKey.NAME, migrationUser.getName());
-            propertiesMap.put(JsonKey.ORG_EXT_ID, migrationUser.getOrgExternalId());
-            propertiesMap.put(JsonKey.UPDATED_ON, new Timestamp(System.currentTimeMillis()));
-            propertiesMap.put(JsonKey.USER_STATUS,getInputStatus(migrationUser.getInputStatus()));
-            if (shadowUser.getClaimStatus() != ClaimStatus.CLAIMED.getValue() && shadowUser.getClaimStatus() != ClaimStatus.REJECTED.getValue() && shadowUser.getClaimStatus() != ClaimStatus.FAILED.getValue()) {
+    private void updateUserInShadowDb(String processId,MigrationUser migrationUser, ShadowUser existingShadowUser) {
+        if(!isSame(existingShadowUser,migrationUser)){
```
Comment (indrajra) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ShadowUserMigrationScheduler.java:237]: This condition will still fail - here;s the scenario.
1. User A was uploaded and A rejected the claim unknowningly.
2. Admin reuploaded.
isSame = true and therefore !isSame = false.

Comment (amit-tarento) [inline actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ShadowUserMigrationScheduler.java:237]: what if user data is same for REJECTED, FAILED and ORGEXTIDMISMATCH claim status.

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ShadowUserMigrationScheduler.java (hunk 1 above)
Comment (Hari-stackroute) [reply actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ShadowUserMigrationScheduler.java:234]: done

File: actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ShadowUserMigrationScheduler.java (hunk 2 above)
Comment (Hari-stackroute) [reply actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ShadowUserMigrationScheduler.java:237]: now checking status only for claimed-users.

Comment (Hari-stackroute) [reply actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ShadowUserMigrationScheduler.java:237]: now checking status only for claimed-users.

**Pull request merged**